include/glow/Runtime/Provisioner/Provisioner.h

```diff
@@ -67,8 +67,12 @@ class Provisioner final {
   std::vector<DeviceManager *> devices_;
 
   /// Helper function to cleanup a provision call. On failure free the
-  /// compiledFunctions that were created.
-  void cleanupProvision(llvm::ArrayRef<std::string> names, bool failure = true);
+  /// compiledFunctions that were created and remove and networks already added
+  /// to devices.
+  void
+  cleanupProvision(llvm::ArrayRef<std::string> names,
+                   std::map<DeviceIDTy, std::vector<std::string>> deviceCleanup,
```

Comment:
const ref for the map?

Reply:
Also, tiny nit, I'd suggest different naming for `deviceCleanup` -- device cleanup is more what it's used for, right? It's more like the `currentNetworkResidency` or something.